### PR TITLE
Task8

### PR DESCRIPTION
**Pull request by bluluke**
I left the body parser code in the first 6 lines of app.js as the test just hung if I used 

           app.use(express.json)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -257,5 +257,87 @@ describe('POST /api/articles/:article_id/comments', () => {
     })
   });
 });
+describe('PATCH /api/articles/:article_id', () => { 
+  test('201: responds with updated article', () => {
```
Comment (pippaamy) [inline __tests__/app.test.js]: this description needs to be 200

File: __tests__/app.test.js [hunk 2]
```diff
@@ -257,5 +257,87 @@ describe('POST /api/articles/:article_id/comments', () => {
...
+    expect(article.topic).toBe('mitch');
+    expect(article.body).toBe("I find this existence challenging");
+    expect(article.votes).toBe(102);
+    expect(article.article_img_url).toBe("https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700");
+    expect(article.article_id).toBe(1);
+    expect(article.created_at).toBe('2020-07-09T20:11:00.000Z');
+    expect(article.title).toBe("Living in the shadow of a great man");
+   });
+  });
+  test('201: responds with updatedArticle even if request object has additional unnecessary property', () => {
```
Comment (pippaamy) [inline __tests__/app.test.js]: same here

File: db/controllers/articlesController.js [hunk 3]
```diff
@@ -12,4 +12,31 @@ exports.getArticles = (req, res, next) => {
      
             next(err)
         })
+}
+
+exports.patchVote = (req, res, next) => {
+    const incPropInArray = Object.keys(req.body);
```
Comment (pippaamy) [inline db/controllers/articlesController.js]: this is unnecessary and unreliable as the objects are unordered

File: db/controllers/articlesController.js [hunk 4]
```diff
@@ -12,4 +12,31 @@ exports.getArticles = (req, res, next) => {
...
+    const incProp = incPropInArray[0];
+    const incValInArray = Object.values(req.body);
+    const incVal = incValInArray[0];
+    const articleId = Object.values(req.params);
```
Comment (pippaamy) [inline db/controllers/articlesController.js]: you can get anything you need on req.body and req.params

File: endpoints.json [hunk 5]
```diff
@@ -67,7 +67,22 @@
         "created_at": "2020-11-03T21:00:00.000Z"
       }
     ]
+  },
+  "PATCH /api/articles/:article_id": {
+    "description": "Increments votes of given article",
```
Comment (pippaamy) [inline endpoints.json:72]: well remembered :)

File: db/models/articlesModel.js [hunk 6]
```diff
@@ -15,6 +15,27 @@ exports.selectAllArticles = (req) => {
...
+            .then(()=> {
+                return db.query(`SELECT * FROM articles WHERE article_id = 1;`)
+            }).then((result) => {
+                return result.rows;
+            })
+    } else {
+        if(validKey === false) {
+            return Promise.reject({status: 400, msg: "Bad request"});
+        } else {
+        return Promise.reject({status: 404, msg: "Not found"});
```
Comment (pippaamy) [inline db/models/articlesModel.js:38]: no need for this????

**Review by pippaamy — CHANGES_REQUESTED** [review]
great job, some changes to make and then pop in another pr :)

File: db/controllers/articlesController.js (hunk 4 above)
Comment (bluluke) [reply db/controllers/articlesController.js]: Thanks for the feedback. Do I need to check the name of property?

**Pull request closed without merging**